server/src/models/services/schema.js

```diff
@@ -101,8 +113,26 @@ export default function ({ models, loaders }) {
     const has_service = await Service.findOne({ program_ids: program_id });
     return !_.isNull(has_service);
   };
+  const get_service_type_summary = async (lang) => {
+    const services = await Service.find();
+    return _.chain(services)
+      .flatMap(`service_type_${lang}`)
+      .countBy()
+      .map((value, type) => ({
+        id: `${type}_${value}`,
+        [`label_${lang}`]: type,
+        value,
+      }))
+      .value();
+  };
 
   const resolvers = {
+    Gov: {
+      service_summary: (_x, __x, { lang }) => ({
+        id: "service_summary",
+        service_type_summary: get_service_type_summary(lang),
```

Comment:
Could performing this inside a resolver instead of `populate` be a performance bottleneck? It's pretty annoying to compute this inside populate because services data is mostly bilangual string.

Reply:
I can see how doing everything in populate would be annoying. 

At the very least, using dataloaders would help, though. following this pattern you're calling`Model.find()` per aggregation type (you've only got one right now, so it's not a huge deal) but you've got other aggregation types to do (interaction points, top 10 services, etc.) and probably don't want to fetch every single service 6 times per infographic. You've already got the org_id based loader, you can add a government dataloader too. 

Doing this stuff in populate would be safer, though. Is everything really just en/fr strings? In titan we made an effort to give all the lookups a code-friendly identifier called `code`, because sometimes we need to build rules that get triggered when a service is external. We should probably be adding those to the pipeline